poseidon/Cargo.toml

```diff
@@ -33,7 +33,15 @@ ocaml-gen = { workspace = true, optional = true }
 [dev-dependencies]
 serde_json.workspace = true
 hex.workspace = true
+criterion = { version = "0.3", default-features = false, features = [
```

Comment:
Would it work if we put this `criterion` into the top level `Cargo.toml`? Or is this version/feature selection fundamentally incompatible with `criterion` used in different packages within `proof-systems`?